### PR TITLE
Datastore cleanup and bugfix

### PR DESCRIPTION
**Pull request by rustyrussell**

### CODE REVIEW

**Comment by m-schmoock** [comment]
@rustyrussell 
Now the `token->type` of the string is not `JSMN_STRING` anymore but `JSMN_PRIMITIVE` which fails inside the `param_escaped_string` callback, as it epxtects it to also be a string?

**Comment by m-schmoock** [comment]
This 'fixes' the testcases (e.g. `tests/test_plugin.py::test_commando_rune`), but without me knowing what I'm doing there and if that's correct :D

```diff
diff --git a/common/json_param.c b/common/json_param.c
index 7aa217464..f3f350cf8 100644
--- a/common/json_param.c
+++ b/common/json_param.c
@@ -417,7 +417,7 @@ struct command_result *param_escaped_string(struct command *cmd,
                                            const jsmntok_t *tok,
                                            const char **str)
 {
-       if (tok->type == JSMN_STRING) {
+       if (tok->type == JSMN_STRING || JSMN_PRIMITIVE) {
                struct json_escape *esc;
                /* jsmn always gives us ~ well-formed strings. */
                esc = json_escape_string_(cmd, buffer + tok->start,

```

**Comment by rustyrussell** [comment]
> @rustyrussell Now the `token->type` of the string is not `JSMN_STRING` anymore but `JSMN_PRIMITIVE` which fails inside the `param_escaped_string` callback, as it expects it to also be a string?

Yes, the previous code would accept *anything*. But that's clearly a bug, we should insist that a parameter called "string" be a string?

**Comment by m-schmoock** [comment]
> Yes, the previous code would accept _anything_. But that's clearly a bug, we should insist that a parameter called "string" be a string?
> 
> Indeed, fixed.

Yes, thats much better. I din't see that this was caused by the commando plugin sending a u64 'string'.

**Comment by m-schmoock** [comment]
ACK https://github.com/ElementsProject/lightning/pull/5994/commits/6992e4232d1e53559aa3e284aa6fb3b803a87ef1

**Review by endothermicdev — APPROVED** [review]
I like allowing the caller to handle the empty result case.

ACK 6992e4232d1e53559aa3e284aa6fb3b803a87ef1

File: plugins/libplugin.h [hunk 1]
```diff
@@ -310,17 +310,19 @@ void rpc_scan(struct plugin *plugin,
 	      const char *guide,
 	      ...);
 
-/* Helper to scan datastore: can only be used in init callback.  *
- Returns false if field does not exist.  * path is /-separated.  Final
- arg is JSON_SCAN or JSON_SCAN_TAL.
```
Comment (endothermicdev) [inline plugins/libplugin.h:315]: I wouldn't have thought a comment block could give me anxiety, but this one succeeded!

**Pull request merged**